### PR TITLE
feat: add event tracking package for protocol analytics

### PR DESCRIPTION
**Pull request by teslashibe**
This PR implements a package that addresses: https://github.com/masa-finance/masa-oracle/issues/419

## Description
This pull request introduces a new package for event tracking and analytics in the Masa Protocol. The package provides a robust, thread-safe solution for tracking various events, storing them in memory, and sending them to an external API.

## Features
- In-memory event storage with thread-safe operations
- Configurable event sending to external API
- Comprehensive logging using logrus
- Predefined events for common scenarios (work distribution, completion, and failure)
- Customizable event tracking for flexibility

## New Files
- `pkg/event/event_tracker.go`: Core implementation of the EventTracker
- `pkg/event/event_client.go`: Implementation of the EventClient for sending events
- `pkg/event/event_library.go`: Predefined event types and tracking methods
- `pkg/event/README.md`: Documentation for the package

## Usage Example
```go
import "github.com/masa-finance/masa-oracle/pkg/event"

// Create a new event tracker with default config
tracker := event.NewEventTracker(nil)

// Track a custom event
tracker.TrackEvent("custom_event", map[string]interface{}{"key": "value"})

// Use convenience method to track and send a work distribution event
client := event.NewEventClient("https://api.example.com", logger, 10*time.Second)
err := tracker.TrackWorkDistribution(workType, remoteWorker, peerId, client)
if err != nil {
    log.Fatal(err)
}
```

## Todo

- [ ] Update Masa Protocol API to take events 
- [ ] Update config.go to point to Masa Protocol API

## Documentation
- Comprehensive README.md file explaining package usage and features
- Godoc comments for all exported functions and types

**Notes for Reviewers**

@mudler I created an implementation ticket using these event tracking methods in `v0.7.0` which can be reviewed here: https://github.com/masa-finance/masa-oracle/issues/512

- [x] Yes, I signed my commits.

### CODE REVIEW

File: pkg/event/event.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		Data:      data,
+	}
+
+	a.events = append(a.events, event)
+	a.logger.WithFields(logrus.Fields{
+		"event_name": name,
+		"data":       data,
+	}).Info("Event tracked")
+
+	if client != nil {
```
Comment (mudler) [inline pkg/event/event.go:106]: My suggestion here would be to go with something like:

```golang
if client == nil {
   return fmt.Errorf("no client available")
}
return client.SendEvent(event)
```

so we keep the error around if the client isn't there yet

Comment (teslashibe) [reply pkg/event/event.go:106]: Yeah this makes sense @mudler

Comment (teslashibe) [reply pkg/event/event.go:106]: @mudler pushing a commit now

File: pkg/event/event.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		return client.SendEvent(event)
+	} else {
```
Comment (mudler) [inline pkg/event/event.go:117]: small nit(feel free to ignore): this can be condensed in an `else if`

File: pkg/event/event.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+type EventTracker struct {
+	events    []Event
+	mu        sync.Mutex
+	logger    *logrus.Logger
+	config    *Config
+	apiClient *EventClient
+}
+
+func NewEventTracker(config *Config) *EventTracker {
```
Comment (mudler) [inline pkg/event/event.go:38]: since this package seems to be self-contained, would benefit from having unit tests

**Review by mudler — APPROVED** [review]
Just few nits (style/tests missing), but it otherwise looks good enough for a first stab at it

File: pkg/event/event.go (hunk 2 above)
Comment (teslashibe) [reply pkg/event/event.go:117]: lets fix this :)

File: pkg/event/event.go (hunk 3 above)
Comment (teslashibe) [reply pkg/event/event.go:38]: yeah I think it should once @restevens402 goes e2e with a first stab at implementation which is 90% done and should be done tomorrow

**Comment by mudler** [comment]
Superseded by #525

**Pull request closed without merging**